src/test/java/com/bookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,175 @@
+            When POST request is sent to /api/books
+            Then return a Book Dto
+            """)
+    @WithMockUser(username = "admin@bookstore.com", roles = {"ADMIN"})
+    public void save_validRequest_returnBookDto() throws Exception {
+        // Given
+        CreateBookRequestDto createBookRequestDto = new CreateBookRequestDto()
+                .setIsbn(VALID_BOOK_ISBN)
+                .setAuthor(VALID_BOOK_AUTHOR)
+                .setTitle(VALID_BOOK_TITLE)
+                .setPrice(VALID_BOOK_PRICE);
+        BookDto expected = new BookDto()
+                .setIsbn(VALID_BOOK_ISBN)
+                .setAuthor(VALID_BOOK_AUTHOR)
+                .setTitle(VALID_BOOK_TITLE)
+                .setPrice(VALID_BOOK_PRICE);
+        String jsonRequest = objectMapper.writeValueAsString(createBookRequestDto);
+
+        // When
+        MvcResult result = mockMvc.perform(post("/books")
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON_VALUE))
+                .andExpect(status().isOk())
+                .andReturn();
+        BookDto actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                BookDto.class);
+
+        // Then
+        assertTrue(EqualsBuilder.reflectionEquals(expected, actual,
+                "categoryIds", "coverImage", "id", "description"));
+    }
+
+    @Test
+    @DisplayName("""
+            Given a valid role
+            When a GET request is sent to /api/books/
+            Then return a list of Book Dtos
+            """)
+    @WithMockUser(username = "username")
+    @Sql(scripts = {"classpath:/database/books/add-one-category.sql",
+            "classpath:/database/books/add-two-books-with-categories.sql"},
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    public void findAll_validRole_returnListOfBookDtos() throws Exception {
+        // Given
+        List<BookDto> expected = bootstrapBookDtoList();
+
+        // When
+        MvcResult result = mockMvc.perform(get("/books")
+                        .contentType(MediaType.APPLICATION_JSON_VALUE))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        // Then
+        List<BookDto> actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                new TypeReference<>() {
+                });
+        assertEquals(expected.size(), actual.size());
+        for (int i = 0; i < actual.size(); i++) {
+            BookDto actualBook = actual.get(i);
+            BookDto expectedBook = expected.get(i);
+            assertTrue(EqualsBuilder.reflectionEquals(expectedBook, actualBook, "categoryIds",
+                    "coverImage",
+                    "id",
+                    "description",
+                    "price"));
+            assertTrue(compareTruncatedRoundedDecimals(expectedBook.getPrice(),
+                    actualBook.getPrice()));
```

Comment:
I think you are over complicated your tests using this method and  fractional number for price. Please, stick to KISS principle and simplify it. If you want to be sure that logic with book price works correctly add a corresponding tests (sucess/fail) for this